src/rsz/test/cpp/TestBufferRemoval.cc

```diff
@@ -0,0 +1,225 @@
+    odb::dbBox::create(inPin, layer, 0, 0, 10, 10);
+    inPin->setPlacementStatus(odb::dbPlacementStatus::FIRM);
+    // this is needed to create top level STA port object
+    sta_->makePortPin("in1", "input");
+
+    odb::dbBTerm* outPort = odb::dbBTerm::create(out1, "out1");
+    outPort->setIoType(odb::dbIoType::OUTPUT);
+    odb::dbBPin* outPin = odb::dbBPin::create(outPort);
+    layer = block->getTech()->findRoutingLayer(0);
+    odb::dbBox::create(outPin, layer, 990, 990, 1000, 1000);
+    outPin->setPlacementStatus(odb::dbPlacementStatus::FIRM);
+    // why is this step of creating STA port object not automatic after creating
+    // bterm and bpin?
+    sta_->makePortPin("out1", "output");
+
+    odb::dbITerm* inTerm = getFirstInput(b1);
+    inTerm->connect(in1);
+
+    odb::dbITerm* outTerm = b1->getFirstOutput();
+    inTerm = getFirstInput(b2);
+    inTerm->connect(n1);
+    outTerm->connect(n1);
+
+    outTerm = b2->getFirstOutput();
+    inTerm = getFirstInput(b3);
+    inTerm->connect(n2);
+    outTerm->connect(n2);
+
+    outTerm = b3->getFirstOutput();
+    inTerm = getFirstInput(b4);
+    inTerm->connect(n3);
+    outTerm->connect(n3);
+
+    outTerm = b4->getFirstOutput();
+    outTerm->connect(out1);
+
+    // initialize STA
+    sta::Corner* corner = sta_->cmdCorner();
+    sta::PathAPIndex pathAPIndex
+        = corner->findPathAnalysisPt(sta::MinMax::max())->index();
+    sta::Corners* corners = sta_->search()->corners();
+    pathAnalysisPt_ = corners->findPathAnalysisPt(pathAPIndex);
+    sta::Graph* graph = sta_->ensureGraph();
+    sta::Pin* outStaPin = db_network_->dbToSta(outPort);
+    outVertex_ = graph->pinLoadVertex(outStaPin);
+
+    // initializer resizer
+    resizer_ = new rsz::Resizer;
+    stt::SteinerTreeBuilder* stt = new stt::SteinerTreeBuilder;
+    grt::GlobalRouter* grt = new grt::GlobalRouter;
+    dpl::Opendp* dp = new dpl::Opendp;
+    resizer_->init(&logger_, db_.get(), sta_.get(), stt, grt, dp, nullptr);
+  }
+
+  odb::dbITerm* getFirstInput(odb::dbInst* inst) const
+  {
+    odb::dbSet<odb::dbITerm> iterms = inst->getITerms();
+    for (odb::dbITerm* iterm : iterms) {
+      if (iterm->isInputSignal()) {
+        return iterm;
+      }
+    }
+    return nullptr;
+  }
+
+  utl::deleted_unique_ptr<odb::dbDatabase> db_;
+  sta::Unit* power_unit_;
+  std::unique_ptr<sta::dbSta> sta_;
+  sta::LibertyLibrary* library_;
+  utl::Logger logger_;
+  sta::dbNetwork* db_network_;
+  sta::PathAnalysisPt* pathAnalysisPt_;
+  rsz::Resizer* resizer_;
+  sta::Vertex* outVertex_;
+};
+
+TEST_F(BufRemTest, SlackImproves)
+{
+  float origArrival
+      = sta_->vertexArrival(outVertex_, sta::RiseFall::rise(), pathAnalysisPt_);
+
+  // Remove one buffer 'b3' from the buffer chain
+  odb::dbChip* chip = db_->getChip();
+  odb::dbBlock* block = chip->getBlock();
+  odb::dbInst* inst = block->findInst("b2");
+  sta::Instance* sta_inst = db_network_->dbToSta(inst);
+  sta::InstanceSeq* insts = new sta::InstanceSeq;
```

Comment:
You should just use std::make_unique<sta::InstanceSeq>(); that way you don't have to manually delete the pointer.

Reply:
Done